homeassistant/components/hifiberry/media_player.py

```diff
@@ -0,0 +1,222 @@
+    SUPPORT_PAUSE
+    | SUPPORT_VOLUME_SET
+    | SUPPORT_VOLUME_MUTE
+    | SUPPORT_PREVIOUS_TRACK
+    | SUPPORT_NEXT_TRACK
+    | SUPPORT_STOP
+    | SUPPORT_PLAY
+    | SUPPORT_VOLUME_STEP
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+SCAN_INTERVAL = timedelta(seconds=2)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the hifiberry media player platform."""
+
+    data = hass.data[DOMAIN][config_entry.entry_id]
+    audiocontrol2 = data[DATA_HIFIBERRY]
+    meta, volume = data[DATA_INIT]
+    uid = config_entry.entry_id
+    name = f"hifiberry {config_entry.data['host']}"
+
+    entity = HifiberryMediaPlayer(audiocontrol2, meta, volume, uid, name)
+    _LOGGER.debug("Vetadata: %s, Volume: %s", meta, volume)
+    async_add_entities([entity])
+
+
+class HifiberryMediaPlayer(MediaPlayerEntity):
+    """Hifiberry Media Player Object."""
+
+    def __init__(self, audiocontrol2, metadata, volume, uid, name):
+        """Initialize the media player."""
+        self._audiocontrol2 = audiocontrol2
+        self._uid = uid
+        self._name = name
+        self._muted = volume == 0
+        self._volume = self._muted_volume = volume
+        self._metadata = metadata
+        self._available = None
+
+    @property
+    def available(self) -> bool:
+        """Return true if device is responding."""
+        return self._available
+
+    @property
+    def unique_id(self):
+        """Return the unique id for the entity."""
+        return self._uid
+
+    @property
+    def device_info(self):
+        """Return device info for this device."""
+        return {
+            "identifiers": {(DOMAIN, self.unique_id)},
+            "name": self.name,
+            "manufacturer": "Hifiberry",
+        }
+
+    async def async_update(self):
+        """Update state."""
+        try:
+            self._metadata = await self._audiocontrol2.metadata()
+            self._volume = await self._audiocontrol2.volume()
+            LOGGER.debug("Metadata: %s", self._metadata)
+            self._available = True
+        except Audiocontrol2Exception:
+            self._available = False
+
+    @property
+    def media_content_type(self):
+        """Content type of current playing media."""
+        return MEDIA_TYPE_MUSIC
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        status = self._metadata.get("playerState", None)
+        if status == "paused":
+            return STATE_PAUSED
+        if status == "playing":
+            return STATE_PLAYING
+        return STATE_IDLE
+
+    @property
+    def media_position_updated_at(self):
+        """When was the position of the current playing media valid.
+
+        Returns value from homeassistant.util.dt.utcnow().
+        """
+        return self._metadata.get("positionupdate", None)
+
+    @property
+    def media_title(self):
+        """Title of current playing media."""
+        return self._metadata.get("title", None)
```

Comment:
You can omit `None` from all of these, since it is equivalent to calling `get("title")`